src/logs_storage/mod.rs

```diff
@@ -1,149 +1,263 @@
-use std::sync::atomic::Ordering;
-
 use anyhow::Result;
-use atomic_enum::atomic_enum;
 use sqd_messages::{ProstMsg, QueryExecuted};
-use tokio_rusqlite::Connection;
+use tokio_rusqlite::{named_params, Connection};
 
 pub const LOGS_PER_PAGE: usize = 256;
 
 pub struct LogsStorage {
     db: Connection,
-    init_state: AtomicInitState,
 }
 
-#[atomic_enum]
-#[derive(PartialEq)]
-enum InitState {
-    NotInitialized = 0,
-    Initializing = 1,
-    Initialized = 2,
+pub enum LoadedLogs {
+    All(Vec<QueryExecuted>),
+    Partial(Vec<QueryExecuted>),
+}
+
+mod sql {
+    // timestamp is in milliseconds
+    pub const CREATE: &str = r#"
+        BEGIN;
+        CREATE TABLE IF NOT EXISTS query_logs_v2(query_id STRING PRIMARY KEY, timestamp INTEGER, log_msg BLOB);
+        CREATE INDEX IF NOT EXISTS idx_query_logs_v2_timestamp ON query_logs_v2(timestamp);
+        COMMIT;"#;
+
+    pub const CLEANUP: &str = "DELETE FROM query_logs_v2 WHERE timestamp < ?";
+
+    pub const INSERT: &str =
+        "INSERT INTO query_logs_v2(query_id, timestamp, log_msg) VALUES(?, ?, ?)";
+
+    pub const SELECT: &str = r#"
+        SELECT log_msg, query_id
+        FROM query_logs_v2
+        WHERE
+            timestamp >= :timestamp
+            AND (
+                :last_query_id IS NULL
+                OR rowid > (SELECT rowid FROM query_logs_v2 WHERE query_id = :last_query_id)
+            )
+        ORDER BY rowid
```

Comment:
Why `query_id > :last_query_id`? Query IDs are UUIDs, they never go in ascending order.

We can do something like this:
```sql
WHERE
    timestamp > :timestamp
    OR (
        timestamp = :timestamp
        AND query_id > IFNULL(:last_query_id, '')
    )
ORDER BY timestamp, query_id
```

But
- It will not work for collecting up to the latest timestamps where new queries may arrive with older timestamps. Why should we make logs storage implicitly depend on a completely irrelevant fact that we only plan to collect queries up to some past timestamp?
- It will work a bit slower than consecutive reads.

If you don't want sqlite-specific syntax, let's just add a `seq_no` column instead of `rowid`

Reply:
We've agreed to have `(timestamp, query_id)` as a cursor for simplicity